### PR TITLE
PCHR-3259: Use lint staged for pre-commit linters

### PR DESCRIPTION
**Pull request by AkA84**
## Overview
This PR introduce the use of [lint-staged](https://github.com/okonet/lint-staged) to manage pre-commit linters, instead of relying on a manually-written hook

### Set up explanation
```json
{
  "scripts": {
    "precommit": "lint-staged"
  }
}
```
This is [husky](https://github.com/typicode/husky), being told to run `lint-staged` whenever a user makes a commit
```json
{
  "lint-staged": {
    "*.js": ["semistandard --fix", "git add"]
  }
}
```
This is the actual `lint-staged` configuration. Here we're telling it to run the `semistandard` linter in auto-fix mode, and if the linter automatically fixes issues on any of the staged files, it should automatically run `git add` on it.

(note that we don't need to exclude the minified files from the selector because `semistandard` already does that by default)

### Adding missing fields to `package.json`
The `package.json` was missing some basic fields, like "name", "license", etc

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,5 +1,32 @@
 {
+  "name": "CiviHR",
+  "description": "HR software for non-profit",
```
Comment (igorpavlov) [inline package.json]: Was this description confirmed with the PM?

File: package.json [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
+  },
   "semistandard": {
-    "globals": [ "CRM", "inject" ]
+    "globals": [
+      "CRM",
+      "inject"
+    ]
+  },
+  "devDependencies": {
+    "husky": "^0.14.3",
```
Comment (igorpavlov) [inline package.json]: Should not the versions be fixed?

File: package.json (hunk 1 above)
Comment (AkA84) [reply package.json]: I took it from civihr.org (doesn't really matter in the end, this package won't be public anyway)

File: package.json (hunk 2 above)
Comment (AkA84) [reply package.json]: no need for that, since we're using package-lock files

**Pull request merged**